chain/dependencies.go

```diff
@@ -60,7 +61,7 @@ type VM interface {
 
 type Mempool interface {
 	Len(context.Context) int
-	Add(context.Context, []*Transaction)
+	Add(context.Context, []mempool.MempoolItem)
```

Comment:
Usually you should try to avoid the repeat of "mempool.Mempool" -> it is probably safe to say `mempool.Item`.